sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionAsyncClient.java

```diff
@@ -0,0 +1,206 @@
+     * Send a PATCH request via a SocketClient
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @return A Mono containing a HttpResponse object
+     */
+    private Mono<HttpResponse> sendPatchViaSocket(HttpRequest httpRequest) {
+        return Mono.fromCallable(() -> SocketClient.sendPatchRequest(httpRequest));
+    }
+
+    /**
+     * Open a connection based on the HttpRequest URL
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @param progressReporter (Optional) for reporting progress while writing the request body
+     * @return A Mono containing a HttpUrlConnectionResponse object
+     */
+    private Mono<HttpResponse> handleConnection(HttpRequest httpRequest, ProgressReporter progressReporter) {
+        return Mono.defer(() -> {
+            try {
+                // Make connection
+                URL url = httpRequest.getUrl();
+                HttpURLConnection connection = null;
+
+                if (proxyOptions != null) {
+                    InetSocketAddress address = proxyOptions.getAddress();
+                    if (address != null) {
+                        Proxy proxy = new Proxy(Proxy.Type.HTTP, address);
+                        connection = (HttpURLConnection) url.openConnection(proxy);
+
+                        if (proxyOptions.getUsername() != null && proxyOptions.getPassword() != null) {
+                            String authString = proxyOptions.getUsername() + ":" + proxyOptions.getPassword();
+                            String authStringEnc = Base64.getEncoder().encodeToString(authString.getBytes());
+                            connection.setRequestProperty("Proxy-Authorization", "Basic " + authStringEnc);
+                        }
+                    }
+                } else {
+                    connection = (HttpURLConnection) url.openConnection();
+                }
+
+                if (connectionTimeout != null) {
+                    connection.setConnectTimeout((int) connectionTimeout.toMillis());
+                }
+
+                // SetConnectionRequest
+                try {
+                    connection.setRequestMethod(httpRequest.getHttpMethod().toString());
+                } catch (ProtocolException e) {
+                    return FluxUtil.monoError(LOGGER, new RuntimeException(e));
+                }
+                for (HttpHeader header : httpRequest.getHeaders()) {
+                    for (String value : header.getValues()) {
+                        connection.addRequestProperty(header.getName(), value);
+                    }
+                }
+                if (httpRequest.getHttpMethod() != HttpMethod.GET) {
+                    connection.setDoOutput(true);
+                }
+
+                // Write body
+                switch (httpRequest.getHttpMethod()) {
+                    case POST:
+                    case PUT:
+                    case DELETE:
+                        if (httpRequest.getBody() != null) {
```

Comment:
I think it would be better to work with `BinaryData` here than `Flux<ByteBuffer>` as `getBody` will convert non-`Flux<ByteBuffer>` request bodies to `Flux<ByteBuffer>` which could introduce a lot of overhead.

That said, I think this is something that can be deferred until later on once the initial implementation and testing are complete.